src/Crypto/WebAuthn/Operations/Attestation/TPM.hs

```diff
@@ -0,0 +1,589 @@
+  extHasNestedASN1 = const True
+  extEncode = error "Unimplemented: This library does not implement encoding the SubjectAlternativeName extension"
+  extDecode asn1 =
+    first ("Could not decode ASN1 subject-alternative-name extension: " ++) $
+      runParseASN1 decodeSubjectAlternativeName asn1
+    where
+      decodeSubjectAlternativeName :: ParseASN1 SubjectAlternativeName
+      decodeSubjectAlternativeName =
+        do
+          map <- Map.fromList <$> decodeFields
+          -- https://www.trustedcomputinggroup.org/wp-content/uploads/Credential_Profile_EK_V2.0_R14_published.pdf
+          tpmManufacturer <- maybe (fail "manufacturer field not found in subject alternative name") pure $ Map.lookup [2, 23, 133, 2, 1] map
+          tpmModel <- maybe (fail "model field not found in subject alternative name") pure $ Map.lookup [2, 23, 133, 2, 2] map
+          tpmVersion <- maybe (fail "version field not found in subject alternative name") pure $ Map.lookup [2, 23, 133, 2, 3] map
+          pure SubjectAlternativeName {..}
+
+      decodeFields :: ParseASN1 [(OID, Text)]
+      decodeFields = do
+        next <- hasNext
+        if next
+          then do
+            n <- getNext
+            case n of
+              OID oid -> do
+                m <- getNext
+                case m of
+                  ASN1String asnString -> do
+                    let text = decodeUtf8 $ getCharacterStringRawData asnString
+                    fields <- decodeFields
+                    pure ((oid, text) : fields)
+                  _ -> decodeFields
+              _ -> decodeFields
+          else pure []
+
+instance M.AttestationStatementFormat Format where
+  type AttStmt Format = Statement
+
+  asfIdentifier _ = "tpm"
+
+  type AttStmtDecodingError Format = DecodingError
+
+  asfDecode _ xs =
+    case (xs !? "ver", xs !? "alg", xs !? "x5c", xs !? "sig", xs !? "certInfo", xs !? "pubArea") of
+      (Just (CBOR.TString "2.0"), Just (CBOR.TInt algId), Just (CBOR.TList (NE.nonEmpty -> Just x5cRaw)), Just (CBOR.TBytes sig), Just (CBOR.TBytes certInfoRaw), Just (CBOR.TBytes pubAreaRaw)) ->
+        do
+          x5c@(aikCert :| _) <- forM x5cRaw $ \case
+            CBOR.TBytes certBytes ->
+              first DecodingErrorCertificate (X509.decodeSignedCertificate certBytes)
+            _ ->
+              Left (DecodingErrorUnexpectedCBORStructure xs)
+          alg <- maybe (Left $ DecodingErrorUnknownAlgorithmIdentifier algId) Right (toAlg algId)
+          -- The get interface requires lazy bytestrings but we typically use
+          -- strict bytestrings in the library, so we have to convert between
+          -- them
+          certInfo <- bimap (\(bs, bo, st) -> DecodingErrorTPM (toStrict bs, bo, st)) (\(_, _, x) -> x) $ Get.runGetOrFail getTPMAttest (fromStrict certInfoRaw)
+          pubArea <- bimap (\(bs, bo, st) -> DecodingErrorTPM (toStrict bs, bo, st)) (\(_, _, x) -> x) $ Get.runGetOrFail getTPMTPublic (fromStrict pubAreaRaw)
```

Comment:
```suggestion
          (_, _, certInfo) <- first DecodingErrorTPM $ Get.runGetOrFail getTPMAttest (fromStrict certInfoRaw)
          (_, _, pubArea) <- first DecodingErrorTPM $ Get.runGetOrFail getTPMTPublic (fromStrict pubAreaRaw)
```

And have the `DecodingErrorTPM` just use a lazy bytestring, I don't think that's a problem.